scipy/sparse/linalg/_eigen/_svds.py

```diff
@@ -327,13 +337,45 @@ def matmat_XH_X(x):
     u = u[:, ::-1]
     s = s[::-1]
     vh = vh[::-1]
+
     jobu = return_singular_vectors in {True, 'u'}
     jobv = return_singular_vectors in {True, 'vh'}
-    if not transpose:
-        u = u if jobu else None
-        vh = vh @ _herm(eigvec) if jobv else None
+
+    if tol is None or tol <= 0.0:
+        res_tol = np.sqrt(1e-15) * max(A.shape)
```

Comment:
@evgueni-ovtchinnikov Would this tol work if A is single precision?

Reply:
@lobpcg thanks for spotting, will correct